streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableKTableOuterJoin.java

```diff
@@ -60,20 +60,21 @@
         }
     }
 
-    private class KTableKTableOuterJoinProcessor extends AbstractProcessor<K, Change<V1>> {
+    private class KTableKTableOuterJoinProcessor implements TypedProcessor<K, Change<V1>, K, Change<R>> {
 
         private final KTableValueGetter<K, V2> valueGetter;
         private StreamsMetricsImpl metrics;
+        private ProcessorContext<K, Change<R>> context;
 
         KTableKTableOuterJoinProcessor(final KTableValueGetter<K, V2> valueGetter) {
             this.valueGetter = valueGetter;
         }
 
         @Override
-        public void init(final ProcessorContext context) {
-            super.init(context);
+        public void init(final ProcessorContext<K, Change<R>> context) {
+            this.context = context;
             metrics = (StreamsMetricsImpl) context.metrics();
-            valueGetter.init(context);
+            valueGetter.init(new ForwardingDisabledProcessorContext(context));
```

Comment:
Thanks to the new bounds on ValueGetter, the type system makes us prove that we won't allow it to call "forward".